runtime/src/main/java/org/corfudb/runtime/collections/TxBuilder.java

```diff
@@ -100,6 +101,37 @@ TxBuilder update(@Nonnull final String tableName,
         return this;
     }
 
+    /**
+     * Updates the value on the specified key by applying a BiFunction
+     *
+     * @param tableName Table name to perform the update on.
+     * @param key       Key
+     * @param mutation  Function to apply to get the new value
+     * @param arg       Argument to pass to the mutation function
+     * @param <K>       Type of Key.
+     * @param <V>       Type of Value.
+     * @return TxBuilder instance
+     */
+    @Nonnull
+    public <K extends Message, V extends Message, M extends Message>
+    TxBuilder update(@Nonnull final String tableName,
+                     @Nonnull final K key,
+                     @Nonnull BiFunction<V, V, V> mutation,
+                     @Nonnull final V arg,
+                     @Nullable final M metadata) {
+        Table<K, V, M> table = getTable(tableName);
+        V currentVal = table.get(key).getPayload();
+        if (currentVal == null) {
+            throw new IllegalArgumentException(
```

Comment:
This has implications on transaction abort - right now you want the transaction to abort on this illegalArgumentException, so perhaps you want to throw a runtime exception instead that wraps an abortCause?